src/main/java/apoc/coll/Coll.java

```diff
@@ -828,4 +828,42 @@ public boolean different(@Name("values") List<Object> values) {
 		if (values == null) return false;
         return new HashSet(values).size() == values.size();
     }
+
+
+
+    @Procedure
+    @Description("apoc.coll.dropDuplicateNeighbors(list) - remove duplicate consecutive objects in a list")
+    public Stream<ListResult> dropDuplicateNeighbors(@Name("list") List<Object> list){
+        boolean hasSameType = true;
+        List<Object> newList = new ArrayList<>();
+        newList.add(list.get(0));
+        Class first = list.get(0).getClass();
+
+        for (int i = 1; i < list.size(); i++) {
+            Object element = list.get(i);
```

Comment:
Cypher lists are not good for index access.
Just use foreach, you don't use the index anywhere.
And it's fine to have a "last" element.